toolkit/tools/pkg/imagecustomizerlib/customizeoverlay.go

```diff
@@ -21,68 +25,128 @@ func enableOverlays(overlays *[]imagecustomizerapi.Overlay, imageChroot *safechr
-			if err != nil {
-				return err
+		lowerDir := overlay.LowerDir
+		upperDir := overlay.UpperDir
+		workDir := overlay.WorkDir
+		mountDependency := overlay.MountDependency
+
+		if overlay.IsRootfsOverlay {
+			lowerDir = path.Join("/sysroot", overlay.LowerDir)
+			upperDir = path.Join("/sysroot", overlay.UpperDir)
+			workDir = path.Join("/sysroot", overlay.WorkDir)
+			if mountDependency != nil {
+				dep := path.Join("/sysroot", *mountDependency)
+				mountDependency = &dep
 			}
 		}
-		// Construct the argument for each Overlay
-		overlayConfig := fmt.Sprintf(
-			"%s,%s,%s,%s",
-			overlay.LowerDir, overlay.UpperDir, overlay.WorkDir, formattedPartition,
-		)
-		overlayConfigs = append(overlayConfigs, overlayConfig)
-	}
 
-	// Concatenate all overlay configurations with spaces
-	concatenatedOverlays := strings.Join(overlayConfigs, " ")
+		options := fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s", lowerDir, upperDir, workDir)
+
+		// Add any additional options if needed (e.g., x-initrd.mount, x-systemd.requires)
+		if mountDependency != nil {
+			options = fmt.Sprintf("%s,x-systemd.requires=%s", options, *mountDependency)
+		}
+		if overlay.IsRootfsOverlay {
+			options = fmt.Sprintf("%s,x-initrd.mount", options)
+		}
+		if overlay.MountOptions != nil && *overlay.MountOptions != "" {
+			options = fmt.Sprintf("%s,%s", options, *overlay.MountOptions)
+		}
 
-	// Construct the final cmdline argument
-	newArgs := []string{
-		fmt.Sprintf("rd.overlayfs=%s", concatenatedOverlays),
+		// Create the FstabEntry based on the overlay.
+		newEntry := diskutils.FstabEntry{
+			Source:  "overlay",
+			Target:  overlay.MountPoint,
+			FsType:  "overlay",
+			Options: options,
+			Freq:    0,
+			PassNo:  0,
+		}
+
+		updatedEntries = append(updatedEntries, newEntry)
 	}
 
-	bootCustomizer, err := NewBootCustomizer(imageChroot)
+	err = diskutils.WriteFstabFile(updatedEntries, fstabFile)
 	if err != nil {
 		return err
 	}
 
-	err = bootCustomizer.UpdateKernelCommandLineArgs(defaultGrubFileVarNameCmdlineLinux, []string{"rd.overlayfs"},
-		newArgs)
-	if err != nil {
-		return err
+	return nil
+}
+
+func createOverlayDirectories(imageChroot *safechroot.Chroot, overlays []imagecustomizerapi.Overlay) error {
+	for _, overlay := range overlays {
+		dirsToCreate := []string{
+			filepath.Join(imageChroot.RootDir(), overlay.MountPoint),
+			filepath.Join(imageChroot.RootDir(), overlay.UpperDir),
+			filepath.Join(imageChroot.RootDir(), overlay.WorkDir),
+		}
+
+		// Iterate over each directory and create it if it doesn't exist.
+		for _, dir := range dirsToCreate {
+			if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+				return fmt.Errorf("failed to create directory (%s): %w", dir, err)
+			}
+		}
 	}
 
-	err = bootCustomizer.WriteToFile(imageChroot)
+	return nil
+}
+
+func addEquivalencyRule(upperDir string, lowerDir string) error {
+	// Construct the semanage command
+	cmd := exec.Command("semanage", "fcontext", "-a", "-e", upperDir, lowerDir)
```

Comment:
Use the `shell.Execute*` or `shell.NewExecBuilder` functions instead of using `exec` directly. The `shell` functions handle automatic logging.